fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java

```diff
@@ -2811,7 +2815,10 @@ public synchronized boolean cancelFragmentInstance(Types.PPlanFragmentCancelReas
                     return false;
                 }
             }
-            this.hasCanceled = true;
+            if (!this.hasCanceled) {
```

Comment:
line 2796 set `this.hasCanceled = true;` way check again ？

Reply:
if all instance getIsDone is true, hasCanceled will be false